src/mpi/romio/adio/common/ad_aggregate.c

```diff
@@ -247,19 +247,19 @@ void ADIOI_Calc_file_domains(ADIO_Offset * st_offsets, ADIO_Offset
  * (including this one)
  */
 void ADIOI_Calc_my_req(ADIO_File fd, ADIO_Offset * offset_list, ADIO_Offset * len_list,
-                       int contig_access_count, ADIO_Offset
-                       min_st_offset, ADIO_Offset * fd_start,
+                       MPI_Count contig_access_count,
+                       ADIO_Offset min_st_offset, ADIO_Offset * fd_start,
                        ADIO_Offset * fd_end, ADIO_Offset fd_size,
                        int nprocs,
-                       int *count_my_req_procs_ptr,
-                       int **count_my_req_per_proc_ptr,
+                       MPI_Count *count_my_req_procs_ptr,
+                       MPI_Count **count_my_req_per_proc_ptr,
                        ADIOI_Access ** my_req_ptr, MPI_Aint ** buf_idx_ptr)
 /* Possibly reconsider if buf_idx's are ok as int's, or should they be aints/offsets?
    They are used as memory buffer indices so it seems like the 2G limit is in effect */
 {
-    int *count_my_req_per_proc, count_my_req_procs;
+    MPI_Count *count_my_req_per_proc, count_my_req_procs, l;
```

Comment:
For my clarity - count_my_req_procs doesn't really need to be MPI_Count; does it? It's the count of processes that have requests bound to this aggregator? No harm though, I suppose.

Reply:
You are correct, and there can only be 2 billion processes associated with an MPI communicator.  I promoted this (and others like it) to  address compiler warnings.   Could cast to int wherever it is used, though "just cast to int" ten years ago is how we got into this mess today!